src/proc/typifier.rs

```diff
@@ -50,6 +51,8 @@ pub enum ResolveError {
     FunctionReturnsVoid,
     #[error("Type is not found in the given immutable arena")]
     TypeNotFound,
+    #[error("{msg}")]
+    Other { msg: String },
```

Comment:
can we get away without the `Other` loophole? I.e. we could have `IncompatibleOperands` error covering a few cases.

Reply:
Ok, I've made an attempt, maybe you can change exact wording/params if needed :)